tests/examples/examples_ft.cpp

```diff
@@ -0,0 +1,229 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <iostream> // TODO remove
+#include <string>
+#include <regex>
+#include <fstream>
+#include <vector>
+#include <k4a/k4a.h>
+#include <utcommon.h>
+#include <gtest/gtest.h>
+
+const static std::string TEST_TEMP_DIR = "examples_test_temp";
+
+// run the specified executable and record the output to output_path
+// if output_path is empty, just output to stdout
+static void run_and_record_executable(const std::string &executable_path,
```

Comment:
Since this function is just appending the executable_path and args, perhaps it would be cleaner to just have a `const std::string &shell_command argument`.

As an example, if it were truly an executable_path argument, the caller may expect spaces in the path to be valid.